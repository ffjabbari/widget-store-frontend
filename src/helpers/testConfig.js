/** Test data setup/teardown helper hooks */

export const TEST_DATA = {};

export const populateTestDataHook = testData => {
  testData.product = {
    byline: 'Latest generation smart televison 2021',
    date_added: '2020-12-29T08:00:00.000Z',
    description:
      'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Nihil, voluptates enim nam voluptatum, quia doloribus quis sequi, porro consequatur excepturi eos error voluptas nobis modi atque magni commodi amet placeat.',
    discount: '0.00',
    image_url:
      'https://www.lg.com/sg/images/tvs/md05271244/gallery/LG-43LH600T-L.jpg',
    name: 'Samsung XG-900 55 inch LCD TV',
    price: '400.40',
    product_id: 1,
    quantity: 100,
    rating: null,
  };

  testData.product2 = {
    byline: 'A good remote.',
    discount: '0.15',
    image_url:
      'https://www.lg.com/sg/images/tvs/md05271244/gallery/LG-43LH600T-L.jpg',
    name: 'Logitech Singularity Universal Remote',
    price: '40.20',
    product_id: 2,
    rating: null,
    quantity: 1,
  };

  testData.userProfileData = {
    data: {
      user: {
        user_id: 1,
        username: 'widgetLord',
        email: 'foo@gmail.com',
        first_name: 'bob',
        last_name: 'bobert',
        address: '123 Hill Ave.',
        address_line2: null,
        city: 'Big City',
        state: 'Texas',
        postal_code: '33333',
        phone_number: null,
        avatar_url: null,
        orders: [
          {
            order_id: 1,
            order_date: '2021-01-24T08:00:00.000Z',
            status: 'Pending',
          },
        ],
      },
    },
  };

  testData.CustomerInfoFormsValues = {
    first_name: 'Test',
    last_name: 'User',
    email: 'foo@gmail.com',
    address: '123 Main St',
    address_line2: '',
    city: 'Big City',
    state: 'TX',
    postal_code: '39488',
    phone_number: '(555) 555-5555',
    set_default: false,
  };

  testData.cart = {
    items: {
      2: {
        byline: 'The only remote you will need',
        date_added: '2021-01-24T08:00:00.000Z',
        description: 'Lorem ipsum dolor sit, placeat.',
        discount: '0.15',
        image_url:
          'https://cnet3.cbsistatic.com/img/SB-Lc9lni02k-GQeFZlVGII6cpg=/1070x602/2010/08/03/57369690-bb77-11e2-8a8e-0291187978f3/33981226-2-1333-OVR-1.jpg',
        name: 'Logitech Singularity Universal Remote',
        price: '40.20',
        product_id: 2,
        quantity: 2,
        rating: null,
      },
    },
    numCartItems: 1,
    subtotal: '34.17',
  };
};
