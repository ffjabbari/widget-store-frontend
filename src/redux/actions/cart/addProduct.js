/** Add_Product action creator. */

import { ADD_PRODUCT } from '../types';

function addProduct(payload) {
  return { type: ADD_PRODUCT, payload };
}
export default addProduct;
