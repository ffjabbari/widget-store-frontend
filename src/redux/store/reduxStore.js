/*
 * Uses redux-persist to persist state.
 * Uses redux-thunk to allow async operations.
 */

import { composeWithDevTools } from 'redux-devtools-extension';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';

// local imports
import root from '../reducers/root';

const persistConfig = {
  key: 'root',
  storage,
  // allow any new state properties to persist state rehydration
  stateReconciler: autoMergeLevel2,
};

const persistedReducer = persistReducer(persistConfig, root);

export const store = createStore(
  persistedReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export const persistedStore = persistStore(store);

export const testStore = createStore(root);
