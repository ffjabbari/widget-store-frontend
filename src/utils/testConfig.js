/** Test data setup/teardown helper hooks */

export const TEST_DATA = {};

export const populateTestDataHook = testData => {
  testData.product = {
    byline: 'Latest generation smart televison 2021',
    date_added: '2020-12-29T08:00:00.000Z',
    description:
      'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Nihil, voluptates enim nam voluptatum, quia doloribus quis sequi, porro consequatur excepturi eos error voluptas nobis modi atque magni commodi amet placeat.',
    discount: '0.00',
    image_url:
      'https://www.lg.com/sg/images/tvs/md05271244/gallery/LG-43LH600T-L.jpg',
    name: 'Samsung XG-900 55 inch LCD TV',
    price: '400.40',
    product_id: 1,
    quantity: 100,
    rating: null,
  };

  testData.product2 = {
    byline: 'The only remote you will need',
    description: 'Lorem ipsum dolor sit, placeat.',
    discount: '0.15',
    image_url:
      'https://cnet3.cbsistatic.com/img/SB-Lc9lni02k-GQeFZlVGII6cpg=/1070x602/2010/08/03/57369690-bb77-11e2-8a8e-0291187978f3/33981226-2-1333-OVR-1.jpg',
    name: 'Logitech Singularity Universal Remote',
    price: '40.20',
    product_id: 2,
    quantity: 1,
    rating: null,
  };

  testData.userProfileData = {
    data: {
      user: {
        user_id: 1,
        username: 'widgetLord',
        email: 'foo@gmail.com',
        first_name: 'bob',
        last_name: 'bobert',
        address: '123 Hill Ave.',
        address_line2: null,
        city: 'Big City',
        state: 'Texas',
        postal_code: '33333',
        phone_number: null,
        avatar_url: null,
        orders: [
          {
            order_id: 1,
            order_date: '2021-01-24T08:00:00.000Z',
            status: 'Pending',
          },
        ],
      },
    },
  };

  testData.CustomerInfoFormsValues = {
    first_name: 'Test',
    last_name: 'User',
    email: 'foo@gmail.com',
    address: '123 Main St',
    address_line2: 'Apt 5',
    city: 'Big City',
    state: 'TX',
    postal_code: '39488',
    phone_number: '(555) 555-5555',
  };

  testData.cart = {
    items: {
      2: {
        byline: 'The only remote you will need',
        description: 'Lorem ipsum dolor sit, placeat.',
        discount: '0.15',
        image_url:
          'https://cnet3.cbsistatic.com/img/SB-Lc9lni02k-GQeFZlVGII6cpg=/1070x602/2010/08/03/57369690-bb77-11e2-8a8e-0291187978f3/33981226-2-1333-OVR-1.jpg',
        name: 'Logitech Singularity Universal Remote',
        price: '40.20',
        product_id: 2,
        quantity: 1,
        rating: null,
      },
    },
    numCartItems: 1,
    subtotal: '34.17',
  };

  // mock return from /order/:id
  testData.orderGet = {
    customer: 1,
    customer_info: {
      first_name: 'Test',
      last_name: 'User',
      email: 'foo@gmail.com',
      address: '123 Main St',
      address_line2: '',
      city: 'Big City',
      state: 'TX',
      postal_code: '39488',
      phone_number: '(555) 555-5555',
    },

    discount: null,
    items: [
      { 0: { order_id: 2, product_id: 6, quantity: 1 } },
      { 1: { order_id: 2, product_id: 8, quantity: 1 } },
    ],

    order_date: '2021-02-05T08:00:00.000Z',
    order_id: 2,
    shipping_address: null,
    shipping_cost: '12.00',
    shipping_method: {
      details: { cost: '12.00', name: 'USPS ground shipping (3-7 days)' },
      shipping_method: 'usps_ground',
    },

    status: 'Pending',
    subtotal: '784.16',
    tax: '66.65',
    total: '862.81',
    total_items_quantity: 2,
    tracking_number: null,
  };

  TEST_DATA.shippingMethods = {
    usps_ground: {
      cost: '12.00',
      name: 'USPS ground shipping (3-7 days)',
    },
    usps_priority: {
      cost: '16.00',
      name: 'USPS priority shipping (2-4 days)',
    },
    usps_next_day: {
      cost: '25.00',
      name: 'USPS next day delivery (1 day)',
    },
    ups_ground: {
      cost: '16.00',
      name: 'UPS ground shipping (3-6 days)',
    },
    ups_2nd_day: {
      cost: '20.00',
      name: 'UPS second day delivery (2 days)',
    },
    ups_overnight: {
      cost: '28.00',
      name: 'UPS overnight delivery (1 day)',
    },
  };
};
