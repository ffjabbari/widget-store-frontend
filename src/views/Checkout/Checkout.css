/* @media screen and (max-width: 991px) {
  .Checkout1 .cart-col {
    display: none;
  }
} */

/* shippingForm */
select#shipping_method {
  overflow-y: hidden;
  padding-left: 0;
  padding-right: 0;
}

select#shipping_method option {
  padding: 3px 8px;
}

select#shipping_method option:hover {
  background: var(--dark);
  color: var(--light);
}

/* CustomerInfoForms */
.CustomerInfoForms .form-control:disabled,
.form-control[readonly] {
  color: black;
  font-weight: 500;
}

@media screen and (min-width: 992px) {
  .cart-col > div {
    position: sticky;
    top: 10vh;
  }
}
