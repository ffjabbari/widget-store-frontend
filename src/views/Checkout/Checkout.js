import { Container, Row, Col } from 'reactstrap';
import { Route, useHistory } from 'react-router-dom';
import { shallowEqual, useSelector } from 'react-redux';
import { useState, useEffect } from 'react';
import { motion } from 'framer-motion';

// get our fontawesome imports
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faWindowClose } from '@fortawesome/free-solid-svg-icons';

// local imports
import Navbar from '../../components/Navbar/Navbar';
import LoginSignup from '../../components/LoginSignup/LoginSignup';
import ItemsList from '../../components/common/ItemsList/ItemsList';
import CheckoutForms from '../../components/CheckoutForms/CheckoutForms';
import { calculateTax, calculateTotal } from '../../helpers/monies';
import './Checkout.css';

const Checkout = () => {
  // composite order data for checkout steps
  const [orderData, setOrderData] = useState({});
  // disabled flag for disabling itemsList cart adjustment
  const [disabled, setDisabled] = useState(false);
  const cart = useSelector(state => state.cart, shallowEqual);
  const history = useHistory();

  // When cart state changes...
  //
  // Recalculate tax and total based on the changed cart data.
  // Set orderData.shipping to false to disallow navigating to shipping step
  // to force a shipping cost recalculation.
  useEffect(() => {
    setOrderData(orderData => ({
      ...orderData,
      shipping: false,
      // if customer data calculate tax else return 0
      tax: orderData.customer
        ? calculateTax(cart.subtotal, orderData.customer?.state)
        : 0,
      total: calculateTotal(cart.subtotal, orderData.customer?.state),
    }));
  }, [cart]);

  const handleClose = () => {
    history.goBack();
  };

  return (
    <div className="Checkout text-light">
      <Navbar />
      <motion.div
        initial={{ marginTop: '-100vh', opacity: 0 }}
        animate={{ marginTop: '0vh', opacity: 1 }}
        exit={{ marginTop: '100vh', opacity: 0 }}
        transition={{ duration: 0.5 }}
      >
        <Container
          fluid="xl"
          className="dark-gradient1 my-4 px-0 pt-3 rounded position-relative"
        >
          <div className="Panel-close">
            <button className="btn-noStyle" onClick={handleClose}>
              <FontAwesomeIcon icon={faWindowClose} size={'2x'} />
            </button>
          </div>
          <h1>Checkout</h1>
          <Row className="mx-0">
            <Col xs="12" lg="6" className="py-2 py-lg-3 cart-col px-0">
              <ItemsList disabled={disabled} orderData={orderData} />
            </Col>
            <Col xs="12" lg="6" className="px-3 px-sm-4 py-lg-3">
              <CheckoutForms
                setDisabled={setDisabled}
                orderData={orderData}
                setOrderData={setOrderData}
              />
            </Col>
          </Row>
        </Container>
      </motion.div>

      <Route path={`/checkout/login`}>
        <LoginSignup />
      </Route>
    </div>
  );
};

export default Checkout;
