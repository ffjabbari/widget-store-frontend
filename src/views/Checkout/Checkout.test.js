// local imports
import { renderWithStore } from '../../utils/testHelpers';
import { TEST_DATA, populateTestDataHook } from '../../utils/testConfig';
import { testStore } from '../../redux/store/reduxStore';
import addProduct from '../../redux/actions/cart/addProduct';
import Checkout from './Checkout';

beforeAll(() => {
  populateTestDataHook(TEST_DATA);
  testStore.dispatch(addProduct(TEST_DATA.product));
});

test('renders Checkout', () => {
  renderWithStore(<Checkout />);
});

test('snapshot', async () => {
  const { asFragment } = renderWithStore(<Checkout />);
  expect(asFragment()).toMatchSnapshot();
});
