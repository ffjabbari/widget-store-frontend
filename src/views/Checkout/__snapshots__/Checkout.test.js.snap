// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="Checkout text-light"
  >
    <div
      class="sc-bdfBwQ hnBsPZ"
    >
      <nav
        class="Navbar navbar navbar-expand-md navbar-dark"
      >
        <span
          class="navbar-text"
        >
          <a
            class="navbar-brand brand-style"
            href="/shop"
          >
            Widget Store
          </a>
        </span>
        <button
          aria-label="Toggle navigation"
          class="navbar-toggler"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse"
        >
          <ul
            class="mr-auto navbar-nav"
          >
            <span
              class="navbar-text"
            >
              <a
                href="/new"
              >
                New Products
              </a>
            </span>
            <span
              class="navbar-text"
            >
              <a
                href="/deals"
              >
                Deals
              </a>
            </span>
            <li
              class="dropdown nav-item"
            >
              <a
                aria-expanded="false"
                aria-haspopup="true"
                class="dropdown-toggle nav-link"
                href="#"
              >
                More
              </a>
              <div
                aria-hidden="true"
                class="dropdown-menu"
                role="menu"
                tabindex="-1"
              >
                <button
                  class="dropdown-item"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <a
                    href="/user"
                  >
                    Account
                  </a>
                </button>
                <button
                  class="dropdown-item"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <a
                    href="/orders"
                  >
                    Orders
                  </a>
                </button>
                <div
                  class="dropdown-divider"
                  tabindex="-1"
                />
                <button
                  class="dropdown-item"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <a
                    href="/about"
                  >
                    About
                  </a>
                </button>
                <button
                  class="dropdown-item"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <a
                    href="/contact"
                  >
                    Contact
                  </a>
                </button>
                <button
                  class="dropdown-item"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <a
                    href="/faq"
                  >
                    FAQ
                  </a>
                </button>
              </div>
            </li>
          </ul>
          <a
            href="//login"
          >
            Login
          </a>
          <span
            class="navitem navbar-text"
          >
            <span
              class="CartIconBadge position-relative undefined"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-shopping-cart fa-w-18 fa-2x "
                data-icon="shopping-cart"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="cart-count-badge badge-pill badge badge-primary"
              >
                100
                <span
                  class="sr-only"
                >
                  cart items
                </span>
              </span>
            </span>
          </span>
        </div>
      </nav>
    </div>
    <div
      style="margin-top: -100vh; opacity: 0;"
    >
      <div
        class="dark-gradient1 my-4 px-0 pt-3 rounded position-relative container-xl"
      >
        <div
          class="Panel-close"
        >
          <button
            class="btn-noStyle"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-window-close fa-w-16 fa-2x "
              data-icon="window-close"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-83.6 290.5c4.8 4.8 4.8 12.6 0 17.4l-40.5 40.5c-4.8 4.8-12.6 4.8-17.4 0L256 313.3l-66.5 67.1c-4.8 4.8-12.6 4.8-17.4 0l-40.5-40.5c-4.8-4.8-4.8-12.6 0-17.4l67.1-66.5-67.1-66.5c-4.8-4.8-4.8-12.6 0-17.4l40.5-40.5c4.8-4.8 12.6-4.8 17.4 0l66.5 67.1 66.5-67.1c4.8-4.8 12.6-4.8 17.4 0l40.5 40.5c4.8 4.8 4.8 12.6 0 17.4L313.3 256l67.1 66.5z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <h1>
          Checkout
        </h1>
        <div
          class="mx-0 row"
        >
          <div
            class="py-2 py-lg-3 cart-col px-0 col-12 col-lg-6"
          >
            <div>
              <div
                class="sc-gsTCUz gRHslc ItemsListDiv mx-auto text-dark"
              >
                <div
                  class="sc-dlfnbm gnzRnV border-bottom py-1 bg-secondary text-light"
                >
                  <div
                    class="CartItem-name"
                  >
                    Item
                  </div>
                  <div
                    class="CartItem-price"
                  >
                    Price
                  </div>
                  <div
                    class="CartItem-quant-adj"
                  >
                    Quantity
                  </div>
                  <div
                    class="CartItem-total"
                  >
                    Item Total
                  </div>
                </div>
                <div
                  class="CartItem py-4 "
                >
                  <a
                    class="CartItem-name px-1 font-weight-bold"
                    href="/shop/product/1"
                  >
                    Samsung XG-900 55 inch LCD TV
                  </a>
                  <div
                    class="CartItem-price"
                  >
                    $400.40
                  </div>
                  <div
                    class="CartItem-quant-adj pt-4 pt-sm-0"
                  >
                    <span
                      class="mr-2 border quantDisplay badge badge-light"
                      data-testid="quantDisplay"
                    >
                      100
                    </span>
                    <button
                      aria-label="increment button"
                      class="btn-noStyle text-info increment "
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-plus-square fa-w-14 fa-1x "
                        data-icon="plus-square"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        viewBox="0 0 448 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-32 252c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92H92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                          fill="currentColor"
                        />
                      </svg>
                    </button>
                    <button
                      aria-label="decrement button"
                      class="btn-noStyle text-secondary "
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-minus-square fa-w-14 fa-1x "
                        data-icon="minus-square"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        viewBox="0 0 448 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zM92 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H92z"
                          fill="currentColor"
                        />
                      </svg>
                    </button>
                    <button
                      aria-label="remove button"
                      class="btn-noStyle text-dark ml-2 "
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-trash-alt fa-w-14 fa-1x "
                        data-icon="trash-alt"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        viewBox="0 0 448 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                          fill="currentColor"
                        />
                      </svg>
                    </button>
                  </div>
                  <div
                    class="CartItem-total font-weight-bold"
                  >
                    <div>
                      $40040.00
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="sc-hKgILt cozQkG"
              >
                <table
                  class="sc-eCssSg cdvDQm TabulationTable ml-auto"
                >
                  <tbody>
                    <tr>
                      <td
                        class="text-left"
                      >
                        Subtotal
                      </td>
                      <td
                        class="text-right"
                      >
                        <div>
                          $40040.00
                        </div>
                      </td>
                    </tr>
                    <tr>
                      <td
                        class="text-left"
                      >
                        Tax
                      </td>
                      <td
                        class="text-right"
                      >
                        <div />
                      </td>
                    </tr>
                    <tr>
                      <td
                        class="text-left"
                      >
                        Shipping
                      </td>
                      <td
                        class="text-right"
                      >
                        <div />
                      </td>
                    </tr>
                    <tr
                      class="border-top"
                    >
                      <td
                        class="text-left"
                      >
                        Total
                      </td>
                      <td
                        class="text-right"
                      >
                        <div>
                          $40040.00
                        </div>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
          <div
            class="px-3 px-sm-4 py-lg-3 col-12 col-lg-6"
          >
            <div
              class="CheckoutForms text-left pt-2 pt-sm-0"
            >
              <section
                class="mb-3"
              >
                <button
                  class="mb-2 btn-sm py-0 border text-light btn btn-transparent"
                  type="button"
                >
                  Customer Information 
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-down fa-w-10 fa-1x ml-1"
                    data-icon="caret-down"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
                <div
                  class="collapse show"
                >
                  <div
                    class="CustomerInfoForms px-1 px-md-4 pt-2 pb-3 bg-light text-dark rounded"
                  >
                    <div>
                      <a
                        class="btn btn-info btn-sm mb-3 mt-1"
                        href="/checkout/login"
                      >
                        Login / Create Account
                      </a>
                      <span
                        class="pl-3"
                      >
                        <i>
                          - or checkout as guest below
                        </i>
                      </span>
                    </div>
                    <form
                      action="#"
                      class="text-left CustomerInfoForm"
                    >
                      <div
                        class="collapse show"
                      >
                        <h6
                          class="text-info"
                        >
                          Customer Info
                        </h6>
                        <div
                          class="form-row"
                        >
                          <div
                            class="col-sm-6"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class=""
                                for="first_name"
                              >
                                First Name
                              </label>
                              <input
                                autocomplete="given-name"
                                class="form-control form-control-sm undefined"
                                id="first_name"
                                name="first_name"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                          <div
                            class="col-sm-6"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class=""
                                for="last_name"
                              >
                                Last Name
                              </label>
                              <input
                                autocomplete="family-name"
                                class="form-control form-control-sm undefined"
                                id="last_name"
                                name="last_name"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class="form-row"
                        >
                          <div
                            class="col-sm-6"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class=""
                                for="email"
                              >
                                Email
                              </label>
                              <input
                                autocomplete="email"
                                class="form-control form-control-sm undefined"
                                id="email"
                                name="email"
                                type="email"
                                value=""
                              />
                            </div>
                          </div>
                          <div
                            class="col-sm-6"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class=""
                                for="phone_number"
                              >
                                Phone Number
                              </label>
                              <input
                                autocomplete="tel-national"
                                class="form-control form-control-sm undefined"
                                id="phone_number"
                                name="phone_number"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class="form-group"
                        >
                          <label
                            class=""
                            for="address"
                          >
                            Street Address
                          </label>
                          <input
                            autocomplete="address-line1"
                            class="form-control form-control-sm undefined"
                            id="address"
                            name="address"
                            type="text"
                            value=""
                          />
                        </div>
                        <div
                          class="form-row"
                        >
                          <div
                            class="col-sm-6"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class=""
                                for="address_line2"
                              >
                                Address Line Two
                              </label>
                              <input
                                autocomplete="address-line2"
                                class="form-control form-control-sm undefined"
                                id="address_line2"
                                name="address_line2"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                          <div
                            class="col-sm-6"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class=""
                                for="city"
                              >
                                City
                              </label>
                              <input
                                autocomplete="address-level2"
                                class="form-control form-control-sm undefined"
                                id="city"
                                name="city"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class="form-row"
                        >
                          <div
                            class="col-sm-6"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class=""
                                for="state"
                              >
                                State
                              </label>
                              <input
                                autocomplete="address-level1"
                                class="form-control form-control-sm undefined"
                                id="state"
                                name="state"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                          <div
                            class="col-sm-6"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class=""
                                for="postal_code"
                              >
                                Postal Code
                              </label>
                              <input
                                autocomplete="postal-code"
                                class="form-control form-control-sm undefined"
                                id="postal_code"
                                name="postal_code"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                      <button
                        class="form-control rounded-pill mt-0 mt-lg-4 btn btn-primary"
                        type="submit"
                      >
                        Next
                      </button>
                    </form>
                    <form
                      action="#"
                      class="text-left ShippingAddressForm"
                    >
                      <div
                        class="collapse"
                      >
                        <h6
                          class="text-info"
                        >
                          Shipping Address
                        </h6>
                        <div
                          class="form-row"
                        >
                          <div
                            class="col-sm-6"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class=""
                                for="first_name_shipping"
                              >
                                First Name
                              </label>
                              <input
                                class="form-control form-control-sm undefined"
                                disabled=""
                                id="first_name_shipping"
                                name="first_name"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                          <div
                            class="col-sm-6"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class=""
                                for="last_name_shipping"
                              >
                                Last Name
                              </label>
                              <input
                                class="form-control form-control-sm undefined"
                                disabled=""
                                id="last_name_shipping"
                                name="last_name"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class="form-group"
                        >
                          <label
                            class=""
                            for="address_shipping"
                          >
                            Street Address
                          </label>
                          <input
                            class="form-control form-control-sm undefined"
                            disabled=""
                            id="address_shipping"
                            name="address"
                            type="text"
                            value=""
                          />
                        </div>
                        <div
                          class="form-row"
                        >
                          <div
                            class="col-sm-6"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class=""
                                for="address_line2_shipping"
                              >
                                Address Line Two
                              </label>
                              <input
                                class="form-control form-control-sm undefined"
                                disabled=""
                                id="address_line2_shipping"
                                name="address_line2"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                          <div
                            class="col-sm-6"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class=""
                                for="city_shipping"
                              >
                                City
                              </label>
                              <input
                                class="form-control form-control-sm undefined"
                                disabled=""
                                id="city_shipping"
                                name="city"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class="form-row"
                        >
                          <div
                            class="col-sm-6"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class=""
                                for="state_shipping"
                              >
                                State
                              </label>
                              <input
                                class="form-control form-control-sm undefined"
                                disabled=""
                                id="state_shipping"
                                name="state"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                          <div
                            class="col-sm-6"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class=""
                                for="postal_code_shipping"
                              >
                                Postal Code
                              </label>
                              <input
                                class="form-control form-control-sm undefined"
                                disabled=""
                                id="postal_code_shipping"
                                name="postal_code"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class="text-right"
                        >
                          <button
                            class="rounded-pill btn btn-primary"
                            type="submit"
                          >
                            Update Shipping Address
                          </button>
                          <button
                            class="ml-2 btn btn-secondary"
                            type="button"
                          >
                            Cancel
                          </button>
                        </div>
                      </div>
                    </form>
                  </div>
                </div>
              </section>
              <section
                class="mb-3"
              >
                <button
                  class="mb-2 btn-sm py-0 border text-light btn btn-transparent"
                  type="button"
                >
                  Shipping 
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-down fa-w-10 fa-1x ml-1"
                    data-icon="caret-down"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
                <div
                  class="collapse"
                >
                  <h1>
                    Loading...
                  </h1>
                </div>
              </section>
              <section
                class="mb-3"
              >
                <button
                  class="mb-2 btn-sm py-0 border text-light btn btn-transparent"
                  type="button"
                >
                  Payment
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-down fa-w-10 fa-1x ml-1"
                    data-icon="caret-down"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
                <div
                  class="collapse"
                />
              </section>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
